openapi3/schema_merge.go

```diff
@@ -0,0 +1,887 @@
+			return schema, nil
+		}
+		if ap.Schema != nil && ap.Schema.Value != nil {
+			additionalSchemas = append(additionalSchemas, ap.Schema.Value)
+		}
+	}
+
+	if len(additionalSchemas) > 0 {
+		result, err := mergeFields(additionalSchemas)
+		if err != nil {
+			return schema, err
+		}
+		additionalProperties.Schema = &SchemaRef{
+			Value: result,
+		}
+	}
+
+	schema.AdditionalProperties = *additionalProperties
+	return schema, nil
+}
+
+func resolveEnum(values [][]interface{}) ([]interface{}, error) {
+	var nonEmptyEnum [][]interface{}
+	for _, enum := range values {
+		if len(enum) > 0 {
+			nonEmptyEnum = append(nonEmptyEnum, enum)
+		}
+	}
+	var intersection []interface{}
+	if len(nonEmptyEnum) == 0 {
+		return intersection, nil
+	}
+	intersection = findIntersectionOfArrays(nonEmptyEnum)
+	if len(intersection) == 0 {
+		return intersection, errors.New("unable to resolve Enum conflict: intersection of values must be non-empty")
+	}
+	return intersection, nil
+}
+
+func resolvePattern(values []string) string {
+	var pattern strings.Builder
+	for _, p := range values {
+		if len(p) > 0 {
+			if !isPatternResolved(p) {
+				pattern.WriteString(fmt.Sprintf("(?=%s)", p))
+			} else {
+				pattern.WriteString(p)
+			}
+		}
+	}
+	return pattern.String()
+}
+
+func isPatternResolved(pattern string) bool {
+	match, _ := regexp.MatchString(`^\(\?=.+\)$`, pattern)
+	return match
+}
+
+func findMaxValue(values []uint64) uint64 {
+	max := uint64(0)
+	for _, num := range values {
+		if num > max {
+			max = num
+		}
+	}
+	return max
+}
+
+func findMinValue(values []*uint64) *uint64 {
+	dvalues := []uint64{}
+	for _, v := range values {
+		if v != nil {
+			dvalues = append(dvalues, *v)
+		}
+	}
+	if len(dvalues) == 0 {
+		return nil
+	}
+	min := uint64(math.MaxUint64)
+	for _, num := range dvalues {
+		if num < min {
+			min = num
+		}
+	}
+	return Uint64Ptr(min)
+}
+
+func resolveType(schema *Schema, collection *SchemaCollection) (*Schema, error) {
+	types := filterEmptyStrings(collection.Type)
+	if len(types) == 0 {
+		schema.Type = ""
+		return schema, nil
+	}
+	if areTypesNumeric(types) {
+		for _, t := range types {
+			if t == "integer" {
+				schema.Type = "integer"
+				return schema, nil
+			}
+		}
+		schema.Type = "number"
+		return schema, nil
+	}
+	if allStringsEqual(types) {
+		schema.Type = types[0]
+		return schema, nil
+	}
+	return schema, errors.New(TypeErrorMessage)
+}
+
+func areTypesNumeric(types []string) bool {
+	for _, t := range types {
+		if t != "integer" && t != "number" {
+			return false
+		}
+	}
+	return true
+}
+
+func resolveFormat(schema *Schema, collection *SchemaCollection) (*Schema, error) {
+	formats := filterEmptyStrings(collection.Format)
+	if len(formats) == 0 {
+		schema.Format = ""
+		return schema, nil
+	}
+
+	if areFormatsNumeric(formats) {
+		orderMap := make(map[string]int)
+		orderMap[formatInt32] = 1
+		orderMap[formatInt64] = 2
+		orderMap[formatDouble] = 3
```

Comment:
Shouldn't the order be reversed?
According to the OpenAPI Spec, Double is wider than Float:
| Type | Format | Description|
|-- | -- | -- |
|number | float | Floating-point numbers.|
|number | double | Floating-point numbers with double precision.|